tendrl/integrations/gluster/sds_sync/__init__.py

```diff
@@ -0,0 +1,91 @@
+import etcd
+import time
+
+from tendrl.commons.objects.job import Job
+from tendrl.commons import sds_sync
+from tendrl.commons.utils import log_utils as logger
+
+import uuid
+
+
+class GlusterIntegrtaionsSyncThread(sds_sync.StateSyncThread):
+    def run(self):
+        logger.log(
+            "debug",
+            NS.get("publisher_id", None),
+            {"message": "%s running" % self.__class__.__name__}
+        )
+        while not self._complete.is_set():
+            time.sleep(30)
+            try:
+                nodes = NS._int.client.read("/nodes")
+            except etcd.EtcdKeyNotFound:
+                return
+
+            for node in nodes.leaves:
+                node_id = node.key.split('/')[-1]
+                try:
+                    node_context = NS.tendrl.objects.NodeContext(
+                        node_id=node_id
+                    ).load()
+                    tendrl_context = NS.tendrl.objects.TendrlContext(
+                        node_id=node_id
+                    ).load()
+
+                    if node_context.status != "DOWN" or\
+                       tendrl_context.sds_name != "gluster":
+                        continue
+
+                    # check if the bricks of this node are already
+                    # marked as down
+
+                    bricks = NS._int.client.read(
+                        "clusters/{0}/Bricks/all/{1}".format(
+                            tendrl_context.integration_id,
+                            node_context.fqdn
+                        )
+                    )
+
```

Comment:
Only if the cluster is_managed == "yes"  then check for Brick status

https://github.com/Tendrl/gluster-integration/blob/master/tendrl/gluster_integration/sds_sync/__init__.py#L254

Reply:
ack